src/accounts/auth.guard.ts

```diff
@@ -0,0 +1,23 @@
+// auth.guard.ts
+import { Injectable, CanActivate, ExecutionContext } from '@nestjs/common';
+import { Observable } from 'rxjs';
+import { AuthService } from './auth.service';
+
+@Injectable()
+export class AuthGuard implements CanActivate {
+  constructor(private readonly authService: AuthService) {}
+
+  canActivate(
+    context: ExecutionContext,
+  ): boolean | Promise<boolean> | Observable<boolean> {
+    const request = context.switchToHttp().getRequest();
+
+    const authToken = request.headers['auth-token'];
+    const accountId = request.headers['account-id'];
```

Comment:
We should not expose accountId to users like this in the headers. Account Id can be stored directly in the token itself